docs/concepts/what-is-an-event.md

```diff
@@ -86,131 +165,357 @@ It is possible for the following events to still occur:
+| request_id         | string  | definition |
+| attempt            | int32   | definition |
+| last_failure       | Failure | definition |
+
 ### ActivityTaskCompleted
 
 This event type indicates that the Activity Task has completed.
 The SDK client has picked up and successfully completed the Activity Task.
 This event type contains Activity execution results.
 
+| Field Name         | Type       | Definition |
+| ------------------ | ---------- | ---------- |
+| result             | Payloads   | definition |
+| scheduled_event_id | int64      | definition |
+| started_event_id   | int64      | definition |
+| retry_state        | RetryState | definition |
+
 ### ActivityTaskFailed
 
 This event type indicates that the Activity Task has completed.
 The SDK picked up the Activity Task but unsuccessfully completed it.
 This event type contains Activity execution errors.
 
+| Field Name         | Type       | Definition |
+| ------------------ | ---------- | ---------- |
+| failure            | Failure    | definition |
+| scheduled_event_id | int64      | definition |
+| started_event_id   | int64      | definition |
+| retry_state        | RetryState | definition |
+
 ### ActivityTaskTimedOut
 
 This event type indicates that the Activity has timed out according to the Temporal Server, due to the Activity having not completed within the timeout settings.
 
+| Field Name         | Type       | Definition |
+| ------------------ | ---------- | ---------- |
+| failure            | Failure    | definition |
+| scheduled_event_id | int64      | definition |
+| started_event_id   | int64      | definition |
+| retry_state        | RetryState | definition |
+
 ### ActivityTaskCancelRequested
 
 This event type indicates that a request to cancel the Activity has occurred.
 The SDK client will be able to confirm cancellation of an Activity during an Activity heartbeat.
 
+| Field Name                       | Type  | Definition |
+| -------------------------------- | ----- | ---------- |
+| scheduled_event_id               | int64 | definition |
+| workflow_task_completed_event_id | int64 | definition |
+
 ### ActivityTaskCanceled
 
 This event type indicates that the Activity has been canceled.
 
+| Field Name                       | Type     | Definition                                                                                      |
+| -------------------------------- | -------- | ----------------------------------------------------------------------------------------------- |
+| details                          | Payloads | Additional information reported by the Activity upon confirming cancellation.                   |
+| latest_cancel_requested_event_id | int64    | ID of the most recent 'ACTIVITY_TASK_CANCEL_REQUESTED' event which refers to the same Activity. |
+| scheduled_event_id               | int64    | the ID of the 'ACTIVITY_TASK_SCHEDULED' event this cancel confirmation corresponds to.          |
```

Comment:
`cancellation` should be spelled `cancelation` in prose. The two `l`'s is British English and the one singualr `l`, is American English. [Google's style guide](https://developers.google.com/style/spelling)